ci/docker/python-sdist.dockerfile

```diff
@@ -32,5 +32,6 @@ RUN apt-get update -y -q && \
 COPY python/requirements-build.txt \
      /arrow/python/requirements-build.txt
 RUN pip3 install --requirement /arrow/python/requirements-build.txt
+RUN pip3 install build
```

Comment:
We could also add this to python/requirements-build.txt, depending on if we care to differentiate the requirements for just building from creating a source distribution

Reply:
Would you like to address this comment @WillAyd ? Apart from that, I think we're good to go.